src/mlpack/tests/feedforward_network_test.cpp

```diff
@@ -410,4 +410,134 @@ BOOST_AUTO_TEST_CASE(DropConnectNetworkTest)
       (dataset, labels, dataset, labels, 2, 10, 50, 0.2);
 }
 
-BOOST_AUTO_TEST_SUITE_END();
\ No newline at end of file
+/**
+ * Train and evaluate a Maxout network(with a baselayer) with the
+ * specified structure.
+ */
+template<typename MatType = arma::mat>
+void BuildMaxoutNetwork(MatType& trainData,
+                        MatType& trainLabels,
+                        MatType& testData,
+                        MatType& testLabels,
+                        const size_t outputSize,
+                        const size_t maxoutHiddenLayerSize,
+                        const size_t maxoutUnitSize,
+                        const size_t maxEpochs,
+                        const double classificationErrorThreshold)
+{
+ /*
+  *  Construct a feed forward network with trainData.n_rows input nodes,
+  *  maxoutUnitSize hidden nodes and trainLabels.n_rows output nodes. The
+  *  network struct that looks like:
+  *
+  *  Input         Maxout         Output
+  *  Layer         Layer          Layer
```

Comment:
No. Actually I transfer output from maxout layer to a linear layer and this linear layer's output is output probability (on which log-likelihood with softmax is applied). So the final linear layer is actually output layer.